src/xxh3.rs

```diff
@@ -595,13 +596,18 @@ fn hash128_0(secret: &[u8], seed: u64) -> u128 {
 }
 
 fn hash128_1to3(data: &[u8], secret: &[u8], seed: u64) -> u128 {
-    let x1 = data[0] as u32;
-    let x2 = data[data.len() >> 1] as u32;
-    let x3 = *data.last().unwrap() as u32;
-    let x4 = data.len() as u32;
+    let x1: u32 = data[0].try_into().unwrap();
+    let x2: u32 = data[data.len() >> 1].try_into().unwrap();
+    let x3: u32 = (*data.last().unwrap()).try_into().unwrap();
+    let x4: u32 = data.len().try_into().unwrap();
 
     let combined_low = ((x1 << 16) | (x2 << 24) | x3 | (x4 << 8)) as u64;
-    let combined_high = (combined_low as u32).swap_bytes().rotate_left(13) as u64;
+    let combined_high: u64 = u32::try_from(combined_low)
+        .unwrap()
+        .swap_bytes()
+        .rotate_left(13)
+        .try_into()
```

Comment:
Lets instead remove the `as u64` on the `combined_low` line, then you don't need the `u32::try_from` on the next line